### PR TITLE
Updated hardhat functions and added a script to update contracts

### PR DESCRIPTION
**Pull request by banshee**
This is to support upgrading the contracts on testnets instead of just recreating them from scratch every time.

### CODE REVIEW

File: smart-contracts/hardhat.config.ts [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
         hardhat: {
             allowUnlimitedContractSize: false,
             chainId: 1,
             forking: {
-                url: alchemyUrl,
+                url: mainnetUrl,
                 blockNumber: 12865480,
             }
         },
+        ropsten: {
```
Comment (ElliotFriedman) [inline smart-contracts/hardhat.config.ts:28]: We should add another config right here just called `mainnet` that uses the mainnnet private key and url.

File: smart-contracts/scripts/upgrade_contracts.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import * as hardhat from "hardhat";
```
[the remark this answers is not in the document]
Comment (banshee) [reply smart-contracts/scripts/upgrade_contracts.ts:1]: We're going to add the readme next to https://github.com/Sifchain/sifnode/blob/develop/smart-contracts/Deployment.md as Upgrade.md

File: smart-contracts/hardhat.config.ts (hunk 1 above)
Comment (banshee) [reply smart-contracts/hardhat.config.ts:29]: > We should add another config right here just called `mainnet` that uses the mainnnet private key and url.

done

**Review by ElliotFriedman — APPROVED** [review]
LGTM!

**Pull request merged**